storagenode/peer.go

```diff
@@ -282,6 +295,27 @@ func (peer *Peer) Run(ctx context.Context) error {
 	return group.Wait()
 }
 
+func (peer *Peer) exponentialBackoffBootstrap(ctx context.Context, waitInterval, maxWait time.Duration, attempts int, combined error) error {
```

Comment:
Maybe you can reorganize code to use loop instead of recursion? This should simplify code and will remove `attempts` parameter. Also you can think about extracting a generic function to be able to use is in a different places. Something like `exponentialRetry(waitInterval, maxWait time.Duration, func (ctx context.Context) error)`

Reply:
that sounds good! thanks